runner-core/src/test/java/org/apache/apisix/plugin/runner/codec/impl/FlatBuffersDecoderTest.java

```diff
@@ -0,0 +1,127 @@
+
+    @InjectMocks
+    FlatBuffersDecoder flatBuffersDecoder;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    @DisplayName("test empty data")
+    void testEmptyData(CapturedOutput capturedOutput) {
+        byte[] bytes = new byte[]{};
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive empty data"));
+    }
+
+    @Test
+    @DisplayName("test unsupported type")
+    void testUnsupportedType(CapturedOutput capturedOutput) {
+        byte[] bytes = new byte[]{4};
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive unsupported type: 4"));
+    }
+
+    @Test
+    @DisplayName("test error data length(1)")
+    void testErrorDataLength1(CapturedOutput capturedOutput) {
+        // data length is greater than actual length
+        byte[] bytes = new byte[]{1, 0, 0, 3, 0};
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive error data length"));
+    }
+
+    @Test
+    @DisplayName("test error data length(2)")
+    void testErrorDataLength2(CapturedOutput capturedOutput) {
+        // data length equal to 0
+        byte[] bytes = new byte[]{1, 0, 0, 0, 0};
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive error data length"));
+    }
+
+    @Test
+    @DisplayName("test error data length(3)")
+    void testErrorDataLength3(CapturedOutput capturedOutput) {
+        // wrong data content
+        byte[] bytes = new byte[]{1, 0, 0, 1, 0, 1};
+        ByteBuffer buffer = ByteBuffer.wrap(bytes);
+        A6Request result = flatBuffersDecoder.decode(buffer);
+        Assertions.assertEquals(Code.BAD_REQUEST, ((A6ErrRequest) result).getCode());
+        Assertions.assertTrue(capturedOutput.getOut().contains("receive error data length"));
+    }
+
+    @Test
+    @DisplayName("test get body")
+    void testGetBody() {
+        // {"name":"foo", "value":"bar"}
+        byte[] bytes = new byte[]{1, 0, 0, 64, 12, 0, 0, 0, 0, 0, 6, 0, 8, 0, 4, 0, 6, 0, 0, 0, 4, 0, 0, 0, 1, 0, 0, 0, 12, 0, 0, 0, 8, 0, 12, 0, 8, 0, 4, 0, 8, 0, 0, 0, 8, 0, 0, 0, 12, 0, 0, 0, 3, 0, 0, 0, 98, 97, 114, 0, 3, 0, 0, 0, 102, 111, 111, 0};
```

Comment:
Using "{\"name\":\"foo\", \"value\":\"bar\"}".getBytes() directly, is more readable and reasonable.

Reply:
`bytes` is not equal to `"{"name": "foo", "value": "bar"}".getBytes()`, but consists of a Flatbuffer serialized and assembled with a private data protocol.
I've modified the way the data is generated.

Reply:
resolved, pls recheck